src/routes/blog/spyder-6-daniel/+page.md

```diff
@@ -0,0 +1,102 @@
+
+
+![The Spyder Contributions page on GitHub, showing a graph of all contributions over time, and Daniel highlighted as the number two contributor between Feb 1 2023 and the present, after lead maintainer Carlos Cordoba](daniel-contributions.png "Daniel has the second-most contributions to the core Spyder repository since Feb 2023")
+
+## Spyder Editor migration to new plugin API
+
+We introduced a brand new plugin API with Spyder 5, and as part of that refactored most of Spyder's panes and many other of its internal components to be plugins that use it, rather than tightly coupled with Spyder itself.
+That both makes Spyder much more modular and easier to maintain, but also gives external plugins all the same power and flexibility that those Spyder internal panes and components have!
+However, the one major exception to that in Spyder 5 was the Editor, which wasn't migrated yet due to the high level of complexity and the further new features that would need to be added to the plugin API.
+
+Migrating the Editor to use the modular plugin API was the final piece of the puzzle to complete the API migration making Spyder a fully modular, pluggable application, avoid having to maintain two duplicate APIs and allow interoperability with other plugins, such as those being developed for remote development in Spyder 6.
+Therefore, along with its related API additions and improvements, it also was a prerequisite for fully implementing our new remote development architecture.
+Therefore, we prioritized it for Spyder 6, with Daniel taking the lead on implementing it!
+
+Given the size of the task, Daniel split it into a number of steps, each corresponding to a GitHub Pull Request.
+First, [he split](https://github.com/spyder-ide/spyder/pull/20893) the ``CodeEditor``, the main widget for editing code, into a component in charge of code completion and linting, and another one in charge of editing.
+This helped separate those concerns and improve improve overall modularity and maintainability.
+Then, [he refactored the ``EditorStack`` code](https://github.com/spyder-ide/spyder/pull/21194), the widget in charge of displaying multiple tabbed ``CodeEditor`` views, to facilitate the migration.
+Finally, he migrated the Editor itself to the new API in two steps: an [initial migration](https://github.com/spyder-ide/spyder/pull/21353), and a [followup cleanup](https://github.com/spyder-ide/spyder/pull/22005).
+This last step alone took six months of hard work, but it will help to address a lot of technical debt accumulated throughout the years.<br><br>
+
+
+![The Spyder remote kernel status dialog showing a successful connection, next to the IPython Console showing a kernel error](remote-kernel-issue-windows.png "Daniel helped find and solve remote development issues like this one")
+
+## Remote development architecture quality testing
+
+Once the Editor migration was moving along, helped review, test and QA the new remote development architecture.
```

Comment:
```suggestion
This last step alone took six months of hard work, but it will help to address a lot of technical debt accumulated throughout the years.


## Remote development architecture quality testing

![The Spyder remote kernel status dialog showing a successful connection, next to the IPython Console showing a kernel error](remote-kernel-issue-windows.png "Daniel helped find and solve remote development issues like this one")

Once the Editor migration was moving along, helped review, test and QA the new remote development architecture.
```